libraries/lib-components/PluginProvider.h

```diff
@@ -171,7 +171,7 @@ class COMPONENTS_API PluginProvider  /* not final */
 // be declared static so as not to interfere with other providers during link.
 // ----------------------------------------------------------------------------
 #define DECLARE_PROVIDER_ENTRY(name)                  \
-static PluginProvider * name()
+static std::unique_ptr<PluginProvider> name()
```

Comment:
First commit fails to build on macOS
Fix with
```
#include "PluginProvider.h"
```

in AudioUnitWrapper.cpp

Reply:
Also need a `make_unique` in AudioUnitEffectsModule.cpp as for the other factories.

Also:  commit comment has a spelling mistake

Reply:
> First commit fails to build on macOS
> Also need a make_unique in AudioUnitEffectsModule.cpp as for the other factories.

Sorry, I completely forgot to fix that, will do soon